knative-operator/pkg/controller/knativeserving/kourier/kourier.go

```diff
@@ -1,53 +1,108 @@
 package kourier
 
 import (
+	"context"
+
 	mf "github.com/jcrossley3/manifestival"
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var (
-	log       = common.Log.WithName("kourier")
-	manifests = []string{"deploy/resources/kourier/kourier-latest.yaml", "deploy/resources/kourier/kourier-openshift.yaml"}
+	log  = common.Log.WithName("kourier")
+	path = "deploy/resources/kourier/kourier-latest.yaml"
 )
 
 // Apply applies Kourier resources.
 func Apply(instance *servingv1alpha1.KnativeServing, api client.Client) error {
+	instance.Status.MarkDependencyInstalling("Kourier")
+	if err := api.Status().Update(context.TODO(), instance); err != nil {
+		return err
+	}
+
 	log.Info("Installing Kourier Ingress")
-	for _, path := range manifests {
-		if err := apply(instance, api, path); err != nil {
-			log.Error(err, "Failed to apply %s: %v", path, err)
-			return err
+	manifest, err := mf.NewManifest(path, false, api)
+	if err != nil {
+		return err
+	}
+	transforms := []mf.Transformer{mf.InjectNamespace(ingressNamespace(instance.GetNamespace()))}
+	if err := manifest.Transform(transforms...); err != nil {
+		return err
+	}
+	if err := manifest.ApplyAll(); err != nil {
+		return err
+	}
+	if err := checkDeployments(&manifest, instance, api); err != nil {
+		return err
+	}
+	log.Info("Kourier is ready")
+
+	instance.Status.MarkDependenciesInstalled()
+	return api.Status().Update(context.TODO(), instance)
+}
+
+// Check for deployments in knative-serving-ingress
+// This function is copied from knativeserving_controller.go in serving-operator
+func checkDeployments(manifest *mf.Manifest, instance *servingv1alpha1.KnativeServing, api client.Client) error {
+	log.Info("Checking deployments")
+	available := func(d *appsv1.Deployment) bool {
```

Comment:
Put this into a total function please. Doesn't need to be inlined as a closure here.